modules/gdscript/gdscript.cpp

```diff
@@ -732,10 +732,25 @@ bool GDScript::_update_exports(bool *r_err, bool p_recursive_call, PlaceHolderSc
 					} break;
 					case GDScriptParser::ClassNode::Member::SIGNAL: {
 						// TODO: Cache this in parser to avoid loops like this.
-						Vector<StringName> parameters_names;
+						Vector<Pair<StringName, Variant::Type>> parameters_names;
 						parameters_names.resize(member.signal->parameters.size());
 						for (int j = 0; j < member.signal->parameters.size(); j++) {
-							parameters_names.write[j] = member.signal->parameters[j]->identifier->name;
+							const GDScriptParser::ParameterNode &this_parameter = *member.signal->parameters[j];
+							// has signal_arg_type?
+							Variant::Type type = Variant::NIL;
+							if (this_parameter.datatype_specifier && this_parameter.datatype_specifier->type_chain.size() == 1) {
+								const Variant::Type builtin_type = GDScriptParser::get_builtin_type(this_parameter.datatype_specifier->type_chain[0]->name);
+								// is builtin ?
+								// todo: If in `GDScriptParser::parse_signal()` it can be assumed that the type is always a built-in type then the following test is not required.
```

Comment:
```suggestion
								// TODO: If in `GDScriptParser::parse_signal()` it can be assumed that the type is always a built-in type, then the following test is not required.
```